File: InsuraTech.Domain/Events/PolicyCancelledEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using InsuraTech.Domain.Common;

namespace InsuraTech.Domain.Events
{
    public sealed class PolicyCancelledEvent : IDomainEvent
    {
        public Guid EventId { get; } = Guid.NewGuid();
        public Guid Id => EventId;

        public DateTime OccurredOn { get; } = DateTime.UtcNow;
        public string EventType => nameof(PolicyCancelledEvent);

        public Guid PolicyId { get; }
        public string PolicyNumber { get; }
        public string Reason { get; }
        public DateOnly EffectiveDate { get; }

       public PolicyCancelledEvent(Guid policyId, string policyNumber,
        string reason, DateOnly effectiveDate)
    {
        PolicyId = policyId;
        PolicyNumber = policyNumber;
        Reason = reason;
        EffectiveDate = effectiveDate;
    }
    }
}

File: backend_v2/src/main/java/com/insuratech/domain/events/PolicyCancelledEvent.java
package com.insuratech.domain.events;

import com.insuratech.domain.common.IDomainEvent;

import java.time.Instant;

public record PolicyCancelledEvent(String policyId, String reason, Instant occurredOn) implements IDomainEvent {
    public PolicyCancelledEvent(String policyId, String reason) {
        this(policyId, reason, Instant.now());
    }
}
